### PR TITLE
Setup flow

### PR DESCRIPTION
**Pull request by profxj**
Mainly a flow chart of how the current code for pypit_setup 

A few other doc updates too.

### CODE REVIEW

**Comment by profxj** [comment]
Give this a quick look if you can, @tiffanyhsyu .

Thanks!

**Comment by tiffanyhsyu** [comment]
Looks good to me..!

File: doc/flow.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+   |  |  |  |  ++ Generates setupIDs
+   |  |  |  ++ group_dict generated
+   |  |  |  ├── arsort.write_sorted(group_dict, setup_dict)
+   |  |  |  ├── arsort.write_setup(setup_dict)
+
+
+Items
+-----
+
+Items created and carried around::
```
Comment (tiffanyhsyu) [inline doc/flow.rst]: Would it be useful if we had a short description of what information each of these items carries around? Might not be of interest for people using PYPIT just for data reduction, but I think it helps clarify the flow a bit more.

For example, 
settings.argf: parameters for the data reduction, such as the method for combining calibration frames; includes user-specified parameters to overwrite the default from the user-input .pypit file

I see now as I'm trying to describe these items that they're not so easily describe-able. Some of them contain a lot of information...I'm not sure if we plan to have documentation on all of the argflag parameters that the user can tweak, but if we do in the future (I'm happy to help), we can link to it here.

Anyway, everything else looks good to me!

**Comment by tiffanyhsyu** [comment]
I had a minor comment, although it should probably be addressed in some 'documentation' branch that we have...so I am happy for this to be merged!

File: doc/flow.rst (hunk 1 above)
Comment (profxj) [reply doc/flow.rst]: Excellent comment.  I've started doing something quite 
like this in another PR so am going to defer it for now.

**Comment by profxj** [comment]
merging.

**Pull request merged**